libsolidity/analysis/TypeChecker.cpp

```diff
@@ -366,6 +366,16 @@ void TypeChecker::checkContractIllegalOverrides(ContractDefinition const& _contr
 	}
 }
 
+namespace {
+
+bool functionIsInInterface(FunctionDefinition const& _function)
```

Comment:
Ah no, I meant adding a method the FunctionDefinition class.

Reply:
And that function just returns the contract kind of its scope (and assets if it is not a contract or somehow not set).